samples/sample_common/src/sample_utils.cpp

```diff
@@ -800,146 +852,196 @@ mfxStatus CSmplYUVWriter::WriteNextFrame(mfxFrameSurface1 *pSurface)
+                    tmp[idx] = ((mfxU16*)pBuffer)[idx] >> 6;
+                }
 
-                    for (int idx = 0; idx < pInfo.CropW; idx++)
-                    {
-                        tmp[idx] = shortPtr[idx] >> 6;
-                    }
+                MSDK_CHECK_NOT_EQUAL(
+                    fwrite(((const mfxU8*)tmp.data()), 4, pInfo.CropW, dstFile),
+                    pInfo.CropW, MFX_ERR_UNDEFINED_BEHAVIOR);
+            }
+            else
+            {
+                MSDK_CHECK_NOT_EQUAL(
+                    fwrite(pBuffer, 4, pInfo.CropW, dstFile),
+                    pInfo.CropW, MFX_ERR_UNDEFINED_BEHAVIOR);
+            }
+        }
+        return MFX_ERR_NONE;
+    }
+    break;
 
-                    MSDK_CHECK_NOT_EQUAL(
-                        fwrite(&tmp[0], 1, (mfxU32)pInfo.CropW * 2, dstFile),
-                        (mfxU32)pInfo.CropW * 2, MFX_ERR_UNDEFINED_BEHAVIOR);
+    case MFX_FOURCC_Y410:
+#if (MFX_VERSION >= MFX_VERSION_NEXT)
+    case MFX_FOURCC_Y216:
+#endif
+    {
+        mfxU8* pBuffer = pInfo.FourCC == MFX_FOURCC_Y410 ? (mfxU8*)pData.Y410 : (mfxU8*)pData.Y;
+        for (i = 0; i < pInfo.CropH; i++)
+        {
+            MSDK_CHECK_NOT_EQUAL(
+                fwrite(pBuffer + (pInfo.CropY * pData.Pitch + pInfo.CropX * 4) + i * pData.Pitch, 4, pInfo.CropW, dstFile),
+                pInfo.CropW, MFX_ERR_UNDEFINED_BEHAVIOR);
+        }
+        return MFX_ERR_NONE;
+    }
+    break;
+#endif
+    case MFX_FOURCC_P010:
+    case MFX_FOURCC_P210:
+    {
+        for (i = 0; i < pInfo.CropH; i++)
+        {
+            mfxU16* shortPtr = (mfxU16*)(pData.Y + (pInfo.CropY * pData.Pitch + pInfo.CropX) + i * pData.Pitch);
+            if (pInfo.Shift)
+            {
+                // Convert MS-P010 to P010 and write
+                tmp.resize(pData.Pitch);
 
-                }
-                else
+                for (int idx = 0; idx < pInfo.CropW; idx++)
                 {
-                    MSDK_CHECK_NOT_EQUAL(
-                        fwrite(shortPtr, 1, (mfxU32)pInfo.CropW * 2, dstFile),
-                        (mfxU32)pInfo.CropW * 2, MFX_ERR_UNDEFINED_BEHAVIOR);
+                    tmp[idx] = shortPtr[idx] >> 6;
                 }
-            }
 
-            break;
+                MSDK_CHECK_NOT_EQUAL(
+                    fwrite(&tmp[0], 1, (mfxU32)pInfo.CropW * 2, dstFile),
+                    (mfxU32)pInfo.CropW * 2, MFX_ERR_UNDEFINED_BEHAVIOR);
+
+            }
+            else
+            {
+                MSDK_CHECK_NOT_EQUAL(
+                    fwrite(shortPtr, 1, (mfxU32)pInfo.CropW * 2, dstFile),
+                    (mfxU32)pInfo.CropW * 2, MFX_ERR_UNDEFINED_BEHAVIOR);
+            }
         }
-        case MFX_FOURCC_RGB4:
-        case 100: //DXGI_FORMAT_AYUV
-        case MFX_FOURCC_A2RGB10:
-            // Implementation for RGB4 and A2RGB10 in the next switch below
+
+        break;
+    }
+    case MFX_FOURCC_RGB4:
+    case 100: //DXGI_FORMAT_AYUV
```

Comment:
What is this line doing? That's enumeration for MFX_FOURCC, so reference to DXGI looks strange.

Reply:
These references were here. It is ok to let them stay.

Reply:
Why they _were_ here? and what they are used for _now_?